src/rwa/components/table/transactions/group-transaction-details.tsx

```diff
@@ -104,27 +156,17 @@ export function GroupTransactionDetails(props: GroupTransactionDetailsProps) {
 
     const { fields, append, remove } = useFieldArray({
         control,
-        name: 'fees', // Name of the field array in your form
+        name: 'fees',
     });
 
-    const cashAmount = watch('cashAmount');
-    const fixedIncomeAmount = watch('fixedIncomeAmount');
-    const type = watch('type');
-    const fees = watch('fees');
-    const unitPricePercent = calculateUnitPricePercent(
-        cashAmount,
-        fixedIncomeAmount,
-    );
-    const cashBalanceChange = calculateCashBalanceChange(
-        type,
-        cashAmount,
-        fees,
-    );
-
     const onSubmit: SubmitHandler<GroupTransactionFormInputs> = data => {
         onSubmitForm({
             ...data,
-            cashBalanceChange,
+            cashBalanceChange: calculateCashBalanceChange(
```

Comment:
Since we don't want to call watch or useWatch in this component, we instead do this calculation with the submitted data instead of watched values.